### PR TITLE
add emergency operations center

### PR DESCRIPTION
**Pull request by hpistudent72**

### CODE REVIEW

**Review by Dassderdie — APPROVED** [review]
I applied my feedback in 4359a328e45c567f1a2747858ef23c61d460f9de
@hpistudent72 please have a look at it.

As already discussed - the collapse stuff could be added back later when adding the log.

**Review by ClFeSc — CHANGES_REQUESTED** [review]
Mostly minor things and one thing too big for now.

What is up with the file `shared/src/store/action-reducers/utils/get-template.ts`?

**Review by ClFeSc — CHANGES_REQUESTED** [review]
Another problem: I can alarm a group to a deleted transfer point when it's the previously selected target.

**Comment by ClFeSc** [comment]
See also #404 in this context.

File: frontend/src/app/pages/exercises/exercise/shared/emergency-operations-center/send-alarm-group-interface/send-alarm-group-interface.component.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            *appLet="transferPoints$ | async | values as transferPoints"
+            ngbDropdown
+            class="d-inline-block float-end"
+        >
+            Ziel:
+            <button
+                ngbDropdownToggle
+                type="button"
+                class="btn btn-outline-primary btn-sm ms-3"
+                [disabled]="!transferPoints.length"
```
[the remark this answers is not in the document]
Comment (Dassderdie) [reply frontend/src/app/pages/exercises/exercise/shared/emergency-operations-center/send-alarm-group-interface/send-alarm-group-interface.component.html:14]: yes

File: frontend/src/app/pages/exercises/exercise/shared/emergency-operations-center/send-alarm-group-interface/send-alarm-group-interface.component.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                                alarmGroup.name,
+                                alarmGroupVehicle.time
+                            ),
+                            targetTransferPointId: targetTransferPointId!,
+                        }),
+                    ];
+                }
+            )
+        ).then((requests) => {
+            if (requests.every((request) => request.success)) {
```
[the remark this answers is not in the document]
Comment (Dassderdie) [reply frontend/src/app/pages/exercises/exercise/shared/emergency-operations-center/send-alarm-group-interface/send-alarm-group-interface.component.ts:77]: The `proposeAction` should already show an error automatically.

File: shared/src/models/utils/start-points.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { IsIn, IsNumber, IsString, IsUUID, Min } from 'class-validator';
+import { UUID, uuidValidationOptions } from '../../utils';
+import { getCreate } from './get-create';
+
+export class TransferStartPoint {
```
[the remark this answers is not in the document]
Comment (Dassderdie) [reply shared/src/models/utils/start-points.ts:5]: Probably a good idea: `StartPoint`

**Comment by Dassderdie** [comment]
> Another problem: I can alarm a group to a deleted transfer point when it's the previously selected target.

I wanted to display invalid transferPoints in `app-transfer-point-name` with `text-danger` -> there would be at least a visual indicator for it. (#358)
We could also check whether the transferPoint exists before alarming the group (getStateSnapshot).

File: frontend/src/app/pages/exercises/exercise/shared/emergency-operations-center/send-alarm-group-interface/send-alarm-group-interface.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (hpistudent72) [reply frontend/src/app/pages/exercises/exercise/shared/emergency-operations-center/send-alarm-group-interface/send-alarm-group-interface.component.ts:77]: We get an error if one action fails. This also fixes itself when we take care of https://github.com/hpi-sam/digital-fuesim-manv/issues/406

File: frontend/src/app/pages/exercises/exercise/shared/emergency-operations-center/send-alarm-group-interface/send-alarm-group-interface.component.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    public get targetTransferPointId() {
+        return targetTransferPointId;
+    }
+    public set targetTransferPointId(value: UUID | undefined) {
+        targetTransferPointId = value;
+    }
+
+    public sendAlarmGroup(alarmGroup: AlarmGroup) {
+        // TODO: Refactor this into one action (uuid generation is currently not possible in the reducer)
```
[the remark this answers is not in the document]
Comment (hpistudent72) [reply frontend/src/app/pages/exercises/exercise/shared/emergency-operations-center/send-alarm-group-interface/send-alarm-group-interface.component.ts:45]: Added an Issue for this https://github.com/hpi-sam/digital-fuesim-manv/issues/406

**Review by ClFeSc — COMMENTED** [review]
Should

> > Another problem: I can alarm a group to a deleted transfer point when it's the previously selected target.
> 
> I wanted to display invalid transferPoints in `app-transfer-point-name` with `text-danger` -> there would be at least a visual indicator for it. (#358) We could also check whether the transferPoint exists before alarming the group (getStateSnapshot).

be fixed in this PR? Otherwise, I'm happy with it.

**Comment by Dassderdie** [comment]
> Should
> 
> > > Another problem: I can alarm a group to a deleted transfer point when it's the previously selected target.
> > 
> > 
> > I wanted to display invalid transferPoints in `app-transfer-point-name` with `text-danger` -> there would be at least a visual indicator for it. (#358) We could also check whether the transferPoint exists before alarming the group (getStateSnapshot).
> 
> be fixed in this PR? Otherwise, I'm happy with it.

This will be done in another PR.

**Pull request merged**